homeassistant/components/androidtv/media_player.py

```diff
@@ -326,8 +326,13 @@ def __init__(self, aftv, name, apps, turn_on_command,
         self._device = None
         self._device_properties = self.aftv.device_properties
         self._is_volume_muted = None
-        self._unique_id = 'androidtv-{}-{}'.format(
-            name, self._device_properties['serialno'])
+
+        if 'serialno' in self._device_properties:
+            self._unique_id = 'androidtv-{}-{}'.format(
+                name, self._device_properties['serialno'])
```

Comment:
Oh wait, this is not good, the name should never be part of the unique ID, as that means that changing name, changes everything. It should just be serial no.

Reply:
Done. The format is now `androidtv-<serialno>`.

Reply:
Will this cause any trouble for users who currently have this integration setup with the old unique ID format?

Reply:
You don't even need Android TV in it. It only has to be unique among your own entities.

Yes it will cause issues, but so would renaming have caused. Should label it breaking change.

Reply:
And the solution is to go to Configuration > Entity Registry and delete the previously created Android TV entity (and the new one, too?) and restart?

Reply:
I added a Breaking Changes section to the original post. If there's anything you'd like me to change/correct, let me know.

Reply:
Please remove name from the unique ID, you can just do `self._unique_id = self._device_properties['serialno']`

Reply:
Oops, I thought I did that already. Now it's done!